rllib/algorithms/algorithm.py

```diff
@@ -2663,6 +2310,40 @@ def fn(worker):
         if remove_from_eval_env_runners and self.eval_env_runner_group is not None:
             self.eval_env_runner_group.foreach_env_runner(fn, local_env_runner=True)
 
+    @OldAPIStack
+    @staticmethod
+    def from_state(state: Dict) -> "Algorithm":
+        """Recovers an Algorithm from a state object.
+
+        The `state` of an instantiated Algorithm can be retrieved by calling its
+        `get_state` method. It contains all information necessary
+        to create the Algorithm from scratch. No access to the original code (e.g.
```

Comment:
Does this work now also with algorithms that had defined new attributes/methods? If the class is available it should imo.

Reply:
Yeah, I think so. Users can decide to override the `get_state/set_state` APIs to add more stateful stuff to their state-dicts, but the basic functionality (restoring EnvRunners, RLModule, Learner optimizer states, connector pipelines, etc..) works across all algos.